scripts/transaction_builder.js

```diff
@@ -135,22 +164,32 @@ export class TransactionBuilder {
         this.#addScript({ script, value });
     }
 
-    #addScript({ script, value }) {
-        this.#transaction.vout.push(
-            new CTxOut({
-                script: bytesToHex(script),
-                value,
-            })
-        );
-        this.#valueOut += value;
+    #addScript({ script, value, subtractFeeFromAmt = false }) {
+        let out = new CTxOut({
+            script: bytesToHex(script),
+            value,
+        });
+        // if subtractFeeFromAmt has been set do NOT add dust
+        // We would end up with an UTXO with negative value
+        if (this.isDust({ out, value }) && subtractFeeFromAmt) {
+            return;
+        }
```

Comment:
Isn't it better to mark dust as unspendable? There's not much of a reason to spend it if you're going to pay more in fees